### ISSUE TITLE
Missing /opt/conda/bin path for root user

### ISSUE THREAD

**Issue by iyanmv** [jupyter/docker-stacks#492]
**What docker image you are using?**
`jupyter/scipy-notebook`

**What complete docker command do you run to launch the container (omitting sensitive values)?**

```
sudo docker run --user root -d -p 18767:8888 --name x-scipy\
                -e GRANT_SUDO=yes\
                -e http_proxy="http://172.17.0.1:3128"\
                -e https_proxy="http://172.17.0.1:3128" --restart unless-stopped\
                -e NB_UID=1011 -e NB_GID=1011 -v /home/x/notebooks:/home/jovyan/work\
                jupyter/scipy-notebook start-notebook.sh\
                --NotebookApp.password='sha1:xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx'
```

**What steps do you take once the container is running to reproduce the issue?**

1. Connect to the docker container with `sudo docker exec -itu jovyan x-scipy` or open a terminal in the web interface.
2. Try to use pip (without sudo) -> `pip` works works but fails to install packages due to permissions.
3. Run pip with sudo.
4. Log in as root with `sudo su` and try pip -> `pip` command not found.
5. Full path works -> `sudo /opt/conda/bin/pip`

**What do you expect to happen?**

Run pip with root permissions so it is able to install packages.

**What actually happens?**

`pip` command not found

**Possible solutions**

1. Run pip with sudo (or logged as root) using full path -> `sudo /opt/conda/bin/pip`
2. Add /opt/conda/bin to root `$PATH` env variable.

**Comment by parente** [comment]
The fact that pip is not part of the root user PATH is the expected behavior. The `jovyan` user has read/write permission on /opt/conda, so there shouldn't be a need to run pip or conda as root. (In fact, doing so will create a mix of owners and permission in /opt/conda. 

You could try running with `sudo -E`, but I suspect it will still get dropped from the PATH.

**Comment by iyanmv** [comment]
Yeah, that makes sense and when I run the image without `-e GRANT_SUDO=yes` I can install packages with pip and `jovyan` user. The problem is that this way I cannot install system packages with `apt-get`.

I tried with `sudo -E` and as you say it is not in the path.

**Comment by parente** [comment]
With `-e GRANT_SUDO=yes`, the `jovyan` user should be able to do all of the following:

```
sudo apt-get install -y something
conda install something_else
pip install some_other_package
```

Can you run pip/conda as `jovyan` and apt-get with `sudo` or does that not fit your use case?

**Comment by iyanmv** [comment]
Thanks for following this issue @parente :)
If those three commands worked, that'd be all I need. The problem is that when I set `-e GRANT_SUDO=yes` both `pip` and `conda` works but then there is a problem regarding permissions. Could it be something related to setting `-e NB_UID=1011 -e NB_GID=1011`?

**Comment by parente** [comment]
> Could it be something related to setting -e NB_UID=1011 -e NB_GID=1011?

Yes. Now I see the problem. I ran:

```
docker run --user root -d -p 8888:8888 -e GRANT_SUDO=yes -e NB_UID=1011 -e NB_GID=1011 jupyter/base-notebook start-notebook.sh
```

I got the following expected output:

```
Set username to: jovyan
usermod: no changes
Set jovyan UID to: 1011
Set jovyan GID to: 1011
Granting jovyan sudo access
Execute the command: jupyter notebook
[I 12:24:38.327 NotebookApp] Writing notebook server cookie secret to /home/jovyan/.local/share/jupyter/runtime/notebook_cookie_secret
[W 12:24:38.735 NotebookApp] WARNING: The notebook server is listening on all IP addresses and not using encryption. This is not recommended.
[I 12:24:38.784 NotebookApp] JupyterLab alpha preview extension loaded from /opt/conda/lib/python3.6/site-packages/jupyterlab
[I 12:24:38.795 NotebookApp] Serving notebooks from local directory: /home/jovyan
[I 12:24:38.795 NotebookApp] 0 active kernels
[I 12:24:38.795 NotebookApp] The Jupyter Notebook is running at:
[I 12:24:38.796 NotebookApp] http://[all ip addresses on your system]:8888/?token=0cf6e8f786002a0880ed348ad49c82de280ea1d28378331e
[I 12:24:38.796 NotebookApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[C 12:24:38.797 NotebookApp]

    Copy/paste this URL into your browser when you connect for the first time,
    to login with a token:
        http://localhost:8888/?token=0cf6e8f786002a0880ed348ad49c82de280ea1d28378331e
```

I fired up a terminal in the notebook server and tried to run: 

```
pip install arrow
```

I was greeted with:

```
  Downloading arrow-0.10.0.tar.gz (86kB)
    100% |████████████████████████████████| 92kB 871kB/s
Requirement already satisfied: python-dateutil in /opt/conda/lib/python3.6/site-packages (from arrow)
Requirement already satisfied: six>=1.5 in /opt/conda/lib/python3.6/site-packages (from python-dateutil->arrow)
Building wheels for collected packages: arrow
  Running setup.py bdist_wheel for arrow ... done
  Stored in directory: /home/jovyan/.cache/pip/wheels/f0/4d/9b/692f17c2b531f0435ec1a73d6f936a4c0c252f23b66d5755e6
Successfully built arrow
Installing collected packages: arrow
Exception:
Traceback (most recent call last):
  File "/opt/conda/lib/python3.6/site-packages/pip/basecommand.py", line 215, in main
    status = self.run(options, args)
  File "/opt/conda/lib/python3.6/site-packages/pip/commands/install.py", line 342, in run
    prefix=options.prefix_path,
  File "/opt/conda/lib/python3.6/site-packages/pip/req/req_set.py", line 784, in install
    **kwargs
  File "/opt/conda/lib/python3.6/site-packages/pip/req/req_install.py", line 851, in install
    self.move_wheel_files(self.source_dir, root=root, prefix=prefix)
  File "/opt/conda/lib/python3.6/site-packages/pip/req/req_install.py", line 1064, in move_wheel_files
    isolated=self.isolated,
  File "/opt/conda/lib/python3.6/site-packages/pip/wheel.py", line 345, in move_wheel_files
    clobber(source, lib_dir, True)
  File "/opt/conda/lib/python3.6/site-packages/pip/wheel.py", line 316, in clobber
    ensure_dir(destdir)
  File "/opt/conda/lib/python3.6/site-packages/pip/utils/__init__.py", line 83, in ensure_dir
    os.makedirs(path)
  File "/opt/conda/lib/python3.6/os.py", line 220, in makedirs
    mkdir(name, mode)
PermissionError: [Errno 13] Permission denied: '/opt/conda/lib/python3.6/site-packages/arrow'
```

Checking the jovyan permissions and permissions in `/opt/conda` I see:

```
vyan@44e27cd176d2:~$ id jovyan
id=1011(jovyan) gid=1011(users) groups=1011(users)
```

which looks correct and:

```
jovyan@44e27cd176d2:~$ ls -l /opt
total 4
drwsrwsr-x 1 1000 100 4096 Nov 14 13:12 conda
```

which does not. There's a mismatch somewhere in the what the fix-permission script and the start-notebook script are assuming the permissions will be.

**Comment by parente** [comment]
https://github.com/jupyter/docker-stacks/blob/master/base-notebook/start.sh#L39

Quick look / note for later: The fix might be to add the user to a new group and leave users alone so that all paths don't need to be adjusted, though I thought the fix-permissions script was meant to address this issue.

**Comment by ericdill** [comment]
@parente fixed by #525?
